std/concurrency.d

```diff
@@ -2502,14 +2502,20 @@ auto ref initOnce(alias var)(lazy typeof(var) init, Mutex mutex)
     return var;
 }
 
+deprecated("use the overload that takes a `shared(Mutex)`")
+auto ref initOnce(alias var)(lazy typeof(var) init, Mutex mutex)
```

Comment:
> Let's not (yet). 

Ok, how about changing the deprecation to a comment in the docs indicating that the this non-`shared` overload may be deprecated in the future, so to discourage the use in new code.

> Mutex didn't work with shared at all for the longest time

Yeah, it was a shame. Now that https://github.com/dlang/druntime/pull/1728 is merged we should be good, no?

> I'm not even sure how well it fits within the (theoretical) shared model.

@klickverbot Can you take a look at http://dlang.org/phobos-prerelease/core_sync_mutex and the source and check if there's anything that can be improved w.r.t. the correct use of the `shared` type qualifier?